On gentoo 'id' is /usr/bin/id while @coreutils@ is /bin.
diff --git a/scripts/nix-profile.sh.in b/scripts/nix-profile.sh.in
index a5f5227..1dc5379 100644
--- a/scripts/nix-profile.sh.in
+++ b/scripts/nix-profile.sh.in
@@ -2,3 +2,3 @@ if [ -n "$HOME" ] && [ -n "$USER" ]; then
     __savedpath="$PATH"
-    export PATH=@coreutils@
+    export PATH=@coreutils@:@prefix@/bin:@prefix@/../bin
 
